crates/sdk/libs/near/src/jsonrpc.rs

```diff
@@ -0,0 +1,74 @@
+use calimero_sdk::env;
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+
+use crate::{Error, RpcMethod};
+
+pub struct Client {
+    url: String,
+    id: std::cell::RefCell<u64>,
+}
+
+impl Client {
+    pub fn testnet() -> Self {
+        Self::new("https://rpc.testnet.near.org/".to_string())
+    }
+
+    pub fn mainnet() -> Self {
+        Self::new("https://rpc.mainnet.near.org/".to_string())
+    }
+
+    fn new(url: String) -> Self {
+        Self {
+            url,
+            id: std::cell::RefCell::new(0),
+        }
+    }
+
+    pub fn call<M>(&self, method: M) -> Result<M::Response, Error<M::Error>>
+    where
+        M: RpcMethod,
+    {
+        let headers = [("Content-Type", "application/json")];
+
+        *self.id.borrow_mut() += 1;
+        let body = serde_json::to_vec(&Request {
+            jsonrpc: "2.0",
+            id: *self.id.borrow(),
+            method: method.method_name(),
+            params: method.params()?,
+        })?;
+
+        let response = unsafe { env::ext::fetch(&self.url, "POST", &headers, &body) }
+            .map_err(Error::FetchError)?;
+
+        serde_json::from_slice::<Response<M::Response, M::Error>>(&response)?
```

Comment:
rust should be able to infer the types here, 

```suggestion
        serde_json::from_slice::<Response<_, _>>(&response)?
```